crates/testing/tests/consensus_task.rs

```diff
@@ -242,6 +242,319 @@ async fn test_consensus_vote_with_permuted_dac() {
+            ViewSyncFinalizeCertificate2Recv(cert),
+        ],
+        outputs: vec![
+            exact(ViewChange(ViewNumber::new(3))),
+            exact(QuorumProposalValidated(proposals[2].data.clone())),
+            leaf_decided(),
+            quorum_proposal_send(),
+        ],
+        asserts: vec![is_at_view_number(3)],
+    };
+
+    let consensus_state = ConsensusTaskState::<
+        TestTypes,
+        MemoryImpl,
+        SystemContextHandle<TestTypes, MemoryImpl>,
+    >::create_from(&handle)
+    .await;
+
+    let stages = vec![view_1, view_2, view_3];
+
+    inject_consensus_polls(&consensus_state).await;
+    run_test_script(stages, consensus_state).await;
+}
+
+#[cfg(test)]
+#[cfg_attr(
+    async_executor_impl = "tokio",
+    tokio::test(flavor = "multi_thread", worker_threads = 2)
+)]
+#[cfg_attr(async_executor_impl = "async-std", async_std::test)]
+/// Makes sure that, when a valid ViewSyncFinalize certificate is available, the consensus task
+/// will indeed vote if the cert is valid and matches the correct view number.
+async fn test_view_sync_finalize_vote() {
+    async_compatibility_layer::logging::setup_logging();
+    async_compatibility_layer::logging::setup_backtrace();
+
+    let handle = build_system_handle(5).await.0;
+    let quorum_membership = handle.hotshot.memberships.quorum_membership.clone();
+
+    let view_sync_finalize_data: ViewSyncFinalizeData<TestTypes> = ViewSyncFinalizeData {
+        relay: 10,
+        round: ViewNumber::new(10),
+    };
+
+    let mut generator = TestViewGenerator::generate(quorum_membership.clone());
+    let mut proposals = Vec::new();
+    let mut leaders = Vec::new();
+    let mut votes = Vec::new();
+    let mut vids = Vec::new();
+    let mut dacs = Vec::new();
+    for view in (&mut generator).take(2) {
+        proposals.push(view.quorum_proposal.clone());
+        leaders.push(view.leader_public_key);
+        votes.push(view.create_vote(&handle));
+        vids.push(view.vid_proposal.clone());
+        dacs.push(view.da_certificate.clone());
+    }
+
+    // Each call to `take` moves us to the next generated view. We advance to view
+    // 3 and then add the finalize cert for checking there.
+    generator.add_view_sync_finalize(view_sync_finalize_data);
+    for view in (&mut generator).take(1) {
+        proposals.push(view.quorum_proposal.clone());
+        leaders.push(view.leader_public_key);
+        votes.push(view.create_vote(&handle));
+        vids.push(view.vid_proposal.clone());
+        dacs.push(view.da_certificate.clone());
+    }
+
+    let view_1 = TestScriptStage {
+        inputs: vec![QuorumProposalRecv(proposals[0].clone(), leaders[0])],
+        outputs: vec![
+            exact(ViewChange(ViewNumber::new(1))),
+            exact(QuorumProposalValidated(proposals[0].data.clone())),
+            exact(QuorumVoteSend(votes[0].clone())),
+        ],
+        asserts: vec![is_at_view_number(1)],
+    };
+
+    let view_2 = TestScriptStage {
```

Comment:
To fix the comments below, this view should probably just except a `Timeout` event, or we just skip this stage all together (as if the node never saw anything in view 2)

Reply:
In this case, I will skip the view so that the node is in a prior view (`v-1`), but received no information that would allow it to move to the next view (`v`), and thus would reasonably trigger a view sync in view 3, which should then proceed as expected.